### PR TITLE
Issue 3180465 settings widget refactor

### PR DESCRIPTION
**Pull request by mariusz-andrzejewski-droptica**

### CODE REVIEW

File: modules/custom/d_p/d_p.module [hunk 1]
```diff
@@ -237,9 +247,14 @@ function d_p_preprocess_field(&$variables) {
...
-    if ($entity->getEntityTypeId() == 'paragraph') {
-      $helper = new ParagraphModifiersHelper($entity);
-      $variables['heading_tag'] = $helper->getModifier(SettingsWidget::HEADING_TYPE_SETTING_NAME);
+    if ($entity->getEntityTypeId() === 'paragraph') {
+      try {
+        /** @var \Drupal\d_p\Plugin\Field\ConfigurationStorageFieldItemListInterface $setting_field */
+        $setting_field = ConfigurationStorage::getSettingsFieldFromEntity($entity);
+        $variables['heading_tag'] = $setting_field->getSettingValue(ParagraphSettingTypesInterface::HEADING_TYPE_SETTING_NAME);
+      }
+      catch (MissingConfigurationStorageFieldException $exception) {
```
Comment (grzegorz-pietrzak-droptica) [inline modules/custom/d_p/d_p.module:256]: Missing logger

File: modules/custom/d_p/d_p.module [hunk 2]
```diff
@@ -74,7 +70,7 @@ function d_p_preprocess_paragraph(&$variables) {
         continue;
       }
 
-      $settings = json_decode($field->first()->get('value')->getValue());
+      $settings = $field->first()->getValue();
 
       if (!isset($settings->custom_class) || !isset($settings->custom_theme_colors)
         || strpos($settings->custom_class, 'theme-custom') === FALSE
```
Comment (grzegorz-pietrzak-droptica) [inline modules/custom/d_p/d_p.module:76]: Maybe we could handle this in a new way? To avoid this strpos

Comment (mariusz-andrzejewski-droptica) [reply modules/custom/d_p/d_p.module:76]: Yes, this should be doable, I can rework it to use newly introduced methods.

File: modules/custom/d_p/d_p.module (hunk 1 above)
Comment (mariusz-andrzejewski-droptica) [reply modules/custom/d_p/d_p.module:256]: It's a silent catch. As you can see it is checking only if the given entity is a paragraph type or not. We may not want to put so many error logs for the bundles that do not have/do not support a setting field.

File: modules/custom/d_p/d_p.module (hunk 2 above)
Comment (mariusz-andrzejewski-droptica) [reply modules/custom/d_p/d_p.module]: I've reworked this part a little bit, you can review it again

**Review by grzegorz-pietrzak-droptica — APPROVED** [review]

File: modules/custom/d_p/src/Helper/ParagraphModifiersHelper.php [hunk 3]
```diff
@@ -5,6 +5,18 @@
 use Drupal\d_p\Plugin\Field\FieldWidget\SettingsWidget;
 use Drupal\paragraphs\Entity\Paragraph;
 
+/**
+ * Class ParagraphModifiersHelper.
+ *
+ * @package Drupal\d_p\Helper
+ *
+ * @deprecated in droopler:8.x-2.2 and is removed from droopler:8.x-2.3.
```
[the remark this answers is not in the document]
Comment (mariusz-andrzejewski-droptica) [reply modules/custom/d_p/src/Helper/ParagraphModifiersHelper.php:13]: Well actually this is part of the Drupal coding standard, here is the example from Drupal core
image
It was also suggested by the code sniffer.

**Pull request merged**